### PR TITLE
[neteasemusic] Add subtitles for NetEaseMusic

### PR DESCRIPTION
**Pull request by fqj1994**
Separate to 3 PRs per discussion at issue #6649 .

This requires #6985 being merged first, or tests will fail.

### CODE REVIEW

File: youtube_dl/extractor/neteasemusic.py [hunk 1]
```diff
@@ -108,30 +115,87 @@ class NetEaseMusicIE(NetEaseMusicBaseIE):
...
-            (time_stamp, text) for time_stamp, text in re.findall(lyrics_expr, translated)
-        )
-        lyrics = '\n'.join([
-            '%s%s / %s' % (time_stamp, text, translation_ts_dict.get(time_stamp, ''))
-            for time_stamp, text in original_ts_texts
-        ])
-        return lyrics
+        translated_ts_texts = re.findall(lyrics_expr, translated)
+        lyrics_original = '\n'.join(map(self._fix_lyric_timestamp, original_ts_texts)) + '\n' if original_ts_texts else ''
+        lyrics_translated = '\n'.join(map(self._fix_lyric_timestamp, translated_ts_texts)) + '\n' if translated_ts_texts else ''
```
Comment (dstftw) [inline youtube_dl/extractor/neteasemusic.py]: This code duplication may be eliminated.

File: youtube_dl/extractor/neteasemusic.py [hunk 2]
```diff
@@ -164,9 +228,21 @@ def _real_extract(self, url):
...
+        if lyrics_original:
+            ret['subtitles']['origin'] = [{
+                'ext': 'lrc',
+                'data': lyrics_original
+            }]
+        if lyrics_translated:
+            ret['subtitles']['translated'] = [{
+                'ext': 'lrc',
+                'data': lyrics_translated
+            }]
```
Comment (dstftw) [inline youtube_dl/extractor/neteasemusic.py]: As well as this.

File: youtube_dl/extractor/neteasemusic.py [hunk 3]
```diff
@@ -108,30 +115,87 @@ class NetEaseMusicIE(NetEaseMusicBaseIE):
...
+        # While standard LRC requires exact 2 digits
+        # remvoe the 3rd digit if the there're three
+        match_expr = r'\[([0-9]{2}:[0-9]{2}\.[0-9]{2})(.*)\]'
+        match_result = re.match(match_expr, timestamp)
+        if match_result:
+            # This must match as match_expr is exact the first group of lyrics_expr
+            return "[" + match_result.group(1) + "]"
+        else:
+            # return a valid timestamp to avoid other applications' error
+            return "[00:00.00]"
```
Comment (dstftw) [inline youtube_dl/extractor/neteasemusic.py]: I've already suggested using single quotes.

**Comment by fqj1994** [comment]
@dstftw  updated

**Comment by fqj1994** [comment]
ping @dstftw 
any other modification needed?

**Comment by fqj1994** [comment]
ping @dstftw

**Comment by fqj1994** [comment]
ping @dstftw or anyone else?

**Comment by fqj1994** [comment]
ping @dstftw or @jaimeMF

**Comment by fqj1994** [comment]
rebased to the latest master, and tested working.
Any furthur modification needed? @dstftw

**Comment by dstftw** [comment]
Subtitles extraction should be moved to `_get_subtitles` in order to do extraction only when subtitles are requested.

**Pull request closed without merging**